addons/product_images/wizard/product_get_pic_using_barcode.py

```diff
@@ -0,0 +1,219 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+from datetime import timedelta
+import re
+import requests
+from requests.exceptions import ConnectTimeout, HTTPError, ReadTimeout
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.misc import format_datetime
+
+class ProductGetPicUsingBarcode(models.TransientModel):
+    """ Get product pictures using barcode on Google Image
+
+    :raises UserError: If a scheduled task is already triggered
+    :raises UserError: If the user didn't enter API Key or Search Engine ID in the settings
+    :raises UserError: If there isn't any product without a picture and with a barcode selected.
+    """
+    _name = 'product.images.get.pic.barcode'
+    _description = 'Get Picture with Barcode Number'
+
+    @api.model
+    def default_get(self, fields_list):
+        # Check if the cron is already trigger and raise an appropriate UserError
+        cron_id = self.env.ref('product_images.ir_cron_get_pics_from_google').id
+        date = self.env['ir.cron.trigger'].search([('cron_id', '=', cron_id)]).call_at
+        if date:
+            raise UserError(_("Another task is already scheduled to pick the images on Google "
+                              "on %s", format_datetime(self.env, date)))
+
+        default = super().default_get(fields_list)
+
+        google_custom_search_cx = bool(
+            self.env['ir.config_parameter'].get_param('google.custom_search.cx')
+        )
+        google_custom_search_key = bool(
+            self.env['ir.config_parameter'].get_param('google.custom_search.key')
+        )
+        if not google_custom_search_cx or not google_custom_search_key:
+            raise UserError(_("You didn't configure properly your API Key or Search Engine ID."))
+
+        if self._context.get('active_model') == 'product.template':
+            product_ids = self.env['product.template'].browse(
+                self._context.get('active_ids')
+            ).product_variant_ids.filtered(lambda p: not p.image_1920 and p.barcode)
+        else:
+            product_ids = self.env['product.product'].browse(
+                self._context.get('active_ids')
+            ).filtered(lambda p: not p.image_1920 and p.barcode).ids
+        if not product_ids:
+            raise UserError(_("There isn't any product without a picture and with a barcode."))
+
+        nb_product = len(product_ids)
+        if nb_product > 10000:
+            product_ids = product_ids[:10000]
```

Comment:
Yes, but I understand now that the name was badly chosen. I'll rename `nb_product` to `nb_product_selected`. This variable only serves to display a text in the wizard in case that the limit of 10k is reached.